packages/react-dom/src/__tests__/ReactDOMFiberAsync-test.internal.js

```diff
@@ -677,5 +677,43 @@ describe('ReactDOMFiberAsync', () => {
       // Therefore the form should have been submitted.
       expect(formSubmitted).toBe(true);
     });
+
+    it('discrete events commit in sequence', () => {
+      const buttonRef = React.createRef();
+      class Counter extends React.Component {
+        state = {counter: 0};
+        render() {
+          const {counter} = this.state;
+          return (
+            <button
+              ref={buttonRef}
+              onClick={() => this.setState({counter: counter + 1})}>
+              {counter}
+            </button>
+          );
+        }
+      }
+      ReactDOM.render(
+        <ConcurrentMode>
+          <Counter />
+        </ConcurrentMode>,
+        container,
+      );
+      let button = buttonRef.current;
+      // Dispatch a click event on the counter button to increment
+      function dispatchButtonClick() {
+        let firstEvent = document.createEvent('Event');
+        firstEvent.initEvent('click', true, true);
+        button.dispatchEvent(firstEvent);
+      }
```

Comment:
Nit: You can do [`button.click();`](https://github.com/jsdom/jsdom/blob/16d3913eea0360c7757e75f7e266c3873c85b7dd/lib/jsdom/living/nodes/HTMLElement-impl.js#L34-L63) as well 🙂